rust/main/chains/hyperlane-sealevel/src/rpc/client.rs

```diff
@@ -12,58 +10,40 @@ use solana_client::{
 };
 use solana_program::clock::Slot;
 use solana_sdk::{
-    account::Account,
-    commitment_config::CommitmentConfig,
-    compute_budget::ComputeBudgetInstruction,
-    hash::Hash,
-    instruction::{AccountMeta, Instruction},
-    message::Message,
-    pubkey::Pubkey,
-    signature::{Signature, Signer},
-    transaction::Transaction,
+    account::Account, commitment_config::CommitmentConfig, hash::Hash, pubkey::Pubkey,
+    signature::Signature, transaction::Transaction,
 };
 use solana_transaction_status::{
     EncodedConfirmedTransactionWithStatusMeta, TransactionStatus, UiConfirmedBlock,
-    UiReturnDataEncoding, UiTransactionEncoding,
+    UiTransactionEncoding,
 };
 
-use hyperlane_core::{ChainCommunicationError, ChainResult, U256};
-
-use crate::{
-    error::HyperlaneSealevelError, priority_fee::PriorityFeeOracle,
-    tx_submitter::TransactionSubmitter, SealevelKeypair,
-};
-
-const COMPUTE_UNIT_MULTIPLIER_NUMERATOR: u32 = 11;
-const COMPUTE_UNIT_MULTIPLIER_DENOMINATOR: u32 = 10;
-
-const PRIORITY_FEE_MULTIPLIER_NUMERATOR: u64 = 110;
-const PRIORITY_FEE_MULTIPLIER_DENOMINATOR: u64 = 100;
+use hyperlane_core::{rpc_clients::BlockNumberGetter, ChainCommunicationError, ChainResult, U256};
 
-pub struct SealevelTxCostEstimate {
-    compute_units: u32,
-    compute_unit_price_micro_lamports: u64,
-}
+use crate::error::HyperlaneSealevelError;
 
 /// Wrapper struct around Solana's RpcClient
-pub struct SealevelRpcClient(RpcClient);
+#[derive(Clone)]
+pub struct SealevelRpcClient(Arc<RpcClient>);
```

Comment:
Had to add an Arc here because there wasn't a good way to clone `RpcClient`, but we need it for `FallbackProvider`